snuba/consumers/consumer_builder.py

```diff
@@ -237,7 +238,9 @@ def __build_streaming_strategy_factory(
                 table_writer,
                 metrics=self.metrics,
                 replacements_producer=(
-                    self.producer if self.replacements_topic is not None else None
+                    self.replacements_producer
+                    if self.replacements_topic is not None
+                    else None
```

Comment:
It should be either a producer instance or None
```suggestion
                    self.replacements_producer
```

Reply:
I think that's what this does: if self.replacements_topic is not None, self.replacements_producer is a valid instance and that gets passed to the field. If it is None, then None is passed to the field.

Reply:
Functionally yes it is the same, but I think it's much more straightforward and less error prone to always define replacements_producer and drop this conditional here.

Reply:
Ah ok, that makes sense.